### DISCUSSION TITLE
height/minHeight for Flex component

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#861] [Q&A] [answered]
I was wondering if the Flex component was missing height/maxHeight/minHeight props? I see that it takes similar width props here: https://github.com/twilio-labs/paste/blob/7a7053766bf4153500061ff5b29499ac91ca4662/packages/paste-core/layout/flex/src/index.tsx#L73-L75

The reason I'm asking is because I'm trying to vertically align a Spinner in a Flexed box with a static height, and TypeScript accepts my height props but doesn't actually propagate them.

**Comment by SiTaggart** [comment, marked as answer]
HI @vnguyen94, it's an interesting suggestion that we might consider.

The current idea with components like Flex and Grid is that they are purely convenience components that only do one job really well. For `Flex`, it does flex box layout really well. If you want more than just that one job, then you probably need `Box` as it comes with all the css props you can hopefully ever need.

We're still learning about the usefulness of convenience components like these via adoption, so this is a useful observation.

For now, I'd just use `Box`, we might add further layout properties to `Flex` at a later date.

**Reply by vankhoawin** [reply]
Yeah I tried a few different combinations of Flex and Box but it wasn't giving me the output I wanted -- a vertically aligned container with a fixed/minimum height. Putting a a fixed height Box inside of Flex vertically-aligned the Box, but not the contents inside the Box.

Frankly I could be flexing incorrectly though 😅

**Reply by SiTaggart** [reply]
Could you do something like https://codesandbox.io/s/flexed-and-centered-spinner-rp0l8?file=/src/index.tsx

**Reply by vankhoawin** [reply]
yeah that works too. since Flex is just a Box under the hood, I was expecting Flex to be a subset of Box and would support the props that would be on Box

**Reply by SiTaggart** [reply]
Yeah, Flex right now is a sub set of Box, in that we expose a few of the base Box props and then supplements those few with Flex specific props.

I don't think we'd ever do a super set where we expose all box props and then supplement those with component specific props. Mainly because we don't want folks modifying the behaviour (style is a behaviour in layout components) outside of primitive components.

Box is your friend right now.
